### ISSUE TITLE
Server tick taking more than 40000 miliseconds according to ModernFix

### ISSUE THREAD

**Issue by avc1657** [pop4959/Chunky#323]
Im playing a modpack (fabric) and after trying to pre-gen with chunky, it works for a couple of minutes (sometimes less than 1 minute, sometimes 5 or more) and then the world becomes completely unfunctional, it doesnt crash, but no blocks i break drop, animals stop moving, everything.

I took a look at the log and modernfix generated a dump. What could be the issue?

The log from Chunky start till the end: https://pastebin.com/1e9uTVDc

Edit: i went to the coordinates where the log says the game tried to access a block entity before it was created and in one i found a spider mobspawner and in the other a waystone.

Edit2: Fabric Chunk Pregenerator mod seems to run just ok, no problems.

**Comment by pop4959** [comment]
The culprit looks to be the MCAFabric mod.

"Server thread" prio=8 Id=99 TIMED_WAITING on java.lang.String@425908cc
	at java.base@17.0.8/jdk.internal.misc.Unsafe.park(Native Method)
	-  waiting on java.lang.String@425908cc
	at java.base@17.0.8/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
	at net.minecraft.class_1255.method_20813(class_1255.java:519)
	at net.minecraft.class_1255.method_18857(class_1255.java:141)
	at net.minecraft.class_3215.redirect$bgg000$c2me-base$beforeAwaitChunk(class_3215.java:5639)
	at net.minecraft.class_3215.getChunkBlocking(class_3215.java:2788)
	at net.minecraft.class_3215.method_12121(class_3215.java:2684)
	at net.minecraft.class_1937.method_8402(class_1937.java:204)
	at net.minecraft.class_1937.method_22350(class_1937.java:7049)
	at net.minecraft.class_1937.method_8500(class_1937.java:7044)
	at net.minecraft.class_1937.method_8404(class_1937.java:872)
	at fabric.net.mca.util.WorldUtils.spawnEntity(WorldUtils.java:51)
	at fabric.net.mca.entity.VillagerFactory.spawn(VillagerFactory.java:104)
	**_at fabric.net.mca.server.SpawnQueue.tick(SpawnQueue.java:51)
	at fabric.net.mca.server.world.data.VillageManager.tick(VillageManager.java:165)
	at fabric.net.mca.fabric.MCAFabric.lambda$onInitialize$0(MCAFabric.java:50)
	at fabric.net.mca.fabric.MCAFabric$$Lambda$8907/0x00000008015e2a10.onEndTick(Unknown Source)_**
	at net.fabricmc.fabric.api.event.lifecycle.v1.ServerTickEvents.lambda$static$6(ServerTickEvents.java:63)
	at net.fabricmc.fabric.api.event.lifecycle.v1.ServerTickEvents$$Lambda$4096/0x0000000800cf6a80.onEndTick(Unknown Source)
	at net.minecraft.class_3218.handler$del000$fabric-lifecycle-events-v1$endWorldTick(class_3218.java:8240)
	at net.minecraft.class_3218.method_18765(class_3218.java:396)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:897)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:824)
	at net.minecraft.class_1132.method_3748(class_1132.java:105)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:671)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265)
	at net.minecraft.server.MinecraftServer$$Lambda$19764/0x00000008025f95c8.run(Unknown Source)
	at java.base@17.0.8/java.lang.Thread.run(Thread.java:833)

It's doing a blocking chunk get during the server tick, which is what's causing your crash.

Not related to chunky, other pre-generators "working just ok" is most likely just a coincidence.

Please use the [Discord server](https://discord.gg/ZwVJukcNQG) for support if you have any other questions or need further help troubleshooting.

**Comment by pop4959** [comment]
Also I have seen several other people have issues with MCAFabric, seems to be a common crash with their mod.

**Comment by pop4959** [comment]
It looks like they are aware of the issue: https://github.com/Luke100000/minecraft-comes-alive/issues/842

**Comment by pop4959** [comment]
@avc1657 Looks like one of the recent commits to chunky may have been aimed to fix incompatibilities here between the trio of mods (mcafabric, chunky, c2me).

If you're not already done pre-generating, could be worth trying to update that and see if it helps. Let me know if you get any results, thanks.

**Comment by avc1657** [comment]
How do i update? Is it already available on curseforge or modrinth?

**Comment by pop4959** [comment]
> How do i update? Is it already available on curseforge or modrinth?

Log into Github and download the artifact from [Github Actions](https://github.com/pop4959/Chunky/actions) or grab it from [CodeMC](https://ci.codemc.io/view/Author/job/pop4959/job/Chunky/).
